homeassistant/components/dsmr/sensor.py

```diff
@@ -377,38 +385,38 @@ class DSMRSensorEntityDescription(SensorEntityDescription):
     ),
 )
 
-
-def create_mbus_entity(
-    mbus: int, mtype: int, telegram: Telegram
-) -> DSMRSensorEntityDescription | None:
-    """Create a new MBUS Entity."""
-    if mtype == 3 and hasattr(telegram, f"BELGIUM_MBUS{mbus}_METER_READING2"):
-        return DSMRSensorEntityDescription(
-            key=f"mbus{mbus}_gas_reading",
+SENSORS_MBUS_DEVICE_TYPE: dict[int, tuple[DSMRSensorEntityDescription, ...]] = {
+    MbusDeviceType.GAS: (
+        DSMRSensorEntityDescription(
+            key="gas_reading",
```

Comment:
What if the meter has multiple gas mbus devices?
How can we find out which key is which meter?

Reply:
We don't need to find that out here. This is the generic description of all GAS (or water) meters. We will add multiple entities of this type if there are multiple (different unique id's).

We may want to expose different DeviceInfo structures for each mbus device as a sub device of the main dsmr device.

Reply:
Ps. I don't see that we should add that change as part of a version bump of the library.

Reply:
Well currently you can have mbus1_gas_reading and mbus3_gas_reading sensors for example.
And then it's clear which sensor is which meter.
But yea :)

Reply:
This doesn't change that.
Both the new and old code is based on the translation key water_meter_reading or gas_meter_reading. So the translated entity name and entity_id would not reflect the mbus index

It is only reflected in the unique identifier of the sensors.

We should correct this and make sure the MBUS_EQUIPMENT_IDENTIFER is visible on the (sub) device object, probably as serial number. But not part of this change.